src/main/java/executor/command/CommandAddIncomeReceipt.java

```diff
@@ -0,0 +1,102 @@
+package executor.command;
+
+import executor.task.TaskList;
+import interpreter.Parser;
+import ui.IncomeReceipt;
+import ui.Ui;
+import ui.Wallet;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+public class CommandAddIncomeReceipt extends Command {
+
+    private String userInput;
+    private Double income;
+    private Date date;
+    private ArrayList<String> tags;
+    private String description;
+
+    /**
+     * Constructor for CommandAddIncomeReceipt subCommand Class.
+     * @param userInput The user input from the CLI
+     */
+    public CommandAddIncomeReceipt(String userInput) {
+        this.commandType = CommandType.IN;
+        this.userInput = userInput;
+        this.description = "You can add a new income receipt in format of 'In $5.00 /tags tag'.";
+
+    }
+
+    @Override
+    public void execute(TaskList taskList) {
+    }
+
+    @Override
+    public void execute(Wallet wallet) {
+        parseUserInput();
+        IncomeReceipt r = new IncomeReceipt(this.income, this.date, this.tags);
+    }
+
+    private String parseForTags(String temp) {
+        int indexBackslash = temp.indexOf('/');
+        // Check if '/' exists
+        if (indexBackslash < 0) {
+            return temp;
+        }
+        // Check if any tags exist
+        int indexMsg = temp.indexOf(' ', indexBackslash);
+        if (indexMsg < 0) {
+            return temp;
+        }
+
+        String[] splitDetails = temp.split(temp.substring(indexBackslash, indexMsg), 2);
+        temp = splitDetails[0];
+        if (splitDetails.length > 1) {
+            storeTags(splitDetails[1]);
+        }
+        return temp;
+    }
+
+    public void parseUserInput() {
+        userInput = Parser.removeStr(this.commandType.toString().trim(), userInput);
```

Comment:
`this.` for userInput in the parameters missing. Do you want to create a private method userInput variable here or update the class variable? Whatever it is, it seems confusing and can be improved.